src/fsharp/vs/IncrementalBuild.fs

```diff
@@ -504,197 +505,235 @@ module internal IncrementalBuild =
         
     type Target = Target of INode * int option
 
+    module List =
+        let rec private foldImpl (folder, pending : 'T list, state : 'State) =
+            async {
+            match pending with
+            | [] -> return state
+            | el :: pending ->
+                let! state = folder state el
+                return! foldImpl (folder, pending, state)
+            }
+        
+        let foldAsync (folder : 'State -> 'T -> Async<'State>) (state : 'State) (list : 'T list) : Async<'State> =
+            foldImpl (folder, list, state)
+
     /// Visit each executable action necessary to evaluate the given output (with an optional slot in a
     /// vector output). Call actionFunc with the given accumulator.
-    let ForeachAction ctok (Target(output, optSlot)) bt (actionFunc:Action -> 'T -> 'T) (acc:'T) =
-        let seen = System.Collections.Concurrent.ConcurrentDictionary<Id,bool>()
-        let isSeen id = 
-            if seen.ContainsKey id then true
-            else 
-                seen.[id] <- true
-                false
-                 
-        let shouldEvaluate(bt,currentsig:InputSignature,id) =
-            if currentsig.IsEvaluated then 
-                currentsig <> Signature(bt,id)
-            else false
+    let ForeachAction ctok (Target(output, optSlot)) bt (actionFunc:Action -> 'T -> Async<'T>) (acc:'T) : Async<'T> =
+        async {
```

Comment:
I know you don't like it, but would it be ok to use 2-space indent at least until PR is finalized, to minimize the visible diff?  I know you can add the whitespace option to github, but last time I tried that it inhibited the ability to add code comments in the review.  

Thanks